packages/medusa/src/api/routes/admin/orders/create-swap.ts

```diff
@@ -98,7 +100,13 @@ import { validator } from "../../../../utils/validator"
 export default async (req, res) => {
   const { id } = req.params
 
-  const validated = await validator(AdminPostOrdersOrderSwapsReq, req.body)
+  const validated = req.validatedBody
+  const retrieveConfig = req.retrieveConfig as FindConfig<Order>
+
+  const returnableItemsIndex =
+    retrieveConfig.select?.indexOf("returnable_items") ?? -1
+  const include_returnable_items = returnableItemsIndex !== -1
+  retrieveConfig.select?.splice(returnableItemsIndex, 1)
```

Comment:
It is the same approach as we have done for the total fields, but my first test was having another options in the qp which was `include_returnable_items` but since in the end it add a property on the object I wanted to avoid confusion for the users in regards to the difference between fields and the options. 

In regard to applying it on different endpoint, I did not have done it for this example but if we agree on the approach then all of that will be moved to a middleware which will assign the option on the req filterable fields

Reply:
> will be moved to a middleware which will assign the option on the req filterable fields

This sounds like a solid approach to me. Would it make sense to introduce a separate query param e.g. `include` for these fields?

That way, we could have an `allowedIncludes` property similar to how we approach the relations and fields, and we don't add a property on the object.

Reply:
When I said object I was referring to the order on which the returnable_items are attached to 😄 but yeah we can introduce an include param indeed which would work like fields or relations 👍 I just wanted to avoid confusion between fields and include* as to why for example a user has to add `total` to `fields` and something else to `include`. But at the same time, since the totals fields are disappearing from the `fields` I think it is not too much of a problem. wdyt?

Reply:
Actually, after giving this some thought, I think it is reasonable to add this as an allowed field - and not use a separate property. 

How would the middleware look like for this?

Reply:
I ll do a draft proposal asap